### PR TITLE
Feature/ansible and swarm

### PR DESCRIPTION
**Pull request by mateuszpierzchala-splunk**

### CODE REVIEW

**Comment by mateuszpierzchala-splunk** [comment]
Closes #1774

File: ansible/tasks/deploy_stack.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- name: Copying docker-compose.yml file on the server
+  copy:
+    src: "/opt/tools/app/docker-compose.yml"
+    dest: "/home/{{ ansible_user }}/docker-compose.yml"
+    owner: "{{ ansible_user }}"
+    group: "{{ ansible_user }}"
+    mode: u=rw,g=rw,o=r
+
+
+- name: Create a directory if it does not exist
```
Comment (rjha-splunk) [inline ansible/tasks/deploy_stack.yml:11]: Create local SC4S directory if it doesn't exist

File: ansible/tasks/deploy_stack.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  file:
+    path: /opt/sc4s
+    state: directory
+    mode: '0755'
+
+- name: Create a directory if it does not exist
```
Comment (rjha-splunk) [inline ansible/tasks/deploy_stack.yml:17]: Create SC4S/tls directory if it doesn't exist

File: ansible/tasks/deploy_stack.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  file:
+    path: /opt/sc4s/tls
+    state: directory
+    mode: '0755'
+
+- name: Create a directory if it does not exist
```
Comment (rjha-splunk) [inline ansible/tasks/deploy_stack.yml:23]: Create SC4S/archive directory if it doesn't exist

File: ansible/tasks/deploy_stack.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  file:
+    path: /opt/sc4s/archive
+    state: directory
+    mode: '0755'
+
+- name: Create a directory if it does not exist
```
Comment (rjha-splunk) [inline ansible/tasks/deploy_stack.yml:29]: Create SC4S/local directory if it doesn't exist

File: ansible/tasks/deploy_stack.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  file:
+    path: /opt/sc4s/local
+    state: directory
+    mode: '0755'
+
+- name: Copying env_file file on the server
```
Comment (rjha-splunk) [inline ansible/tasks/deploy_stack.yml:35]: Copying env_file file on the server

File: ansible/tasks/deploy_stack.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  copy:
+    src: "/opt/tools/env_file"
+    dest: "/opt/sc4s/env_file"
+    owner: "{{ ansible_user }}"
+    group: "{{ ansible_user }}"
+    mode: u=rw,g=rw,o=r
+
+- name: Create a volume
```
Comment (rjha-splunk) [inline ansible/tasks/deploy_stack.yml:43]: Create splunk-sc4s-var volume

File: ansible/tasks/deploy_stack.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  docker_volume:
+    name: splunk-sc4s-var
+
+- name: Deploying sc4s app
```
Comment (rjha-splunk) [inline ansible/tasks/deploy_stack.yml:47]: Running docker compose for SC4S

**Review by rjha-splunk — COMMENTED** [review]
Please edit the files as commented.

File: ansible/tasks/deploy_stack.yml (hunk 2 above)
Comment (mateuszpierzchala-splunk) [reply ansible/tasks/deploy_stack.yml]: fixed

File: ansible/tasks/deploy_stack.yml (hunk 4 above)
Comment (mateuszpierzchala-splunk) [reply ansible/tasks/deploy_stack.yml]: fixed

File: ansible/tasks/deploy_stack.yml (hunk 5 above)
Comment (mateuszpierzchala-splunk) [reply ansible/tasks/deploy_stack.yml:35]: fixed.

File: ansible/tasks/deploy_stack.yml (hunk 6 above)
Comment (mateuszpierzchala-splunk) [reply ansible/tasks/deploy_stack.yml]: fixed.

File: ansible/tasks/deploy_stack.yml (hunk 3 above)
Comment (mateuszpierzchala-splunk) [reply ansible/tasks/deploy_stack.yml]: fixed.

File: ansible/tasks/deploy_stack.yml (hunk 7 above)
Comment (mateuszpierzchala-splunk) [reply ansible/tasks/deploy_stack.yml]: fixed.

**Pull request merged**